client-skeleton-subscriber/src/main/java/eu/arrowhead/client/skeleton/subscriber/security/SubscriberNotificationAccessControlFilter.java

```diff
@@ -0,0 +1,119 @@
+package eu.arrowhead.client.skeleton.subscriber.security;
+
+import java.io.IOException;
+import java.security.cert.X509Certificate;
+import java.util.Map;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+
+import eu.arrowhead.client.skeleton.subscriber.constants.SubscriberDefaults;
+import eu.arrowhead.common.CommonConstants;
+import eu.arrowhead.common.Utilities;
+import eu.arrowhead.common.core.CoreSystem;
+import eu.arrowhead.common.exception.ArrowheadException;
+import eu.arrowhead.common.exception.AuthException;
+import eu.arrowhead.common.filter.ArrowheadFilter;
+
+public class SubscriberNotificationAccessControlFilter extends ArrowheadFilter {
+
+	//=================================================================================================
+	// members
+	
+	private static final CoreSystem[] allowedCoreSystemsForSendingNotification = { CoreSystem.EVENT_HANDLER };
+	
+	private Map<String, String> eventTypeMap;
+	private String serverCN;
+
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	public void setEventTypeMap( final Map<String, String> eventTypeMap) { this.eventTypeMap = eventTypeMap;}
+	public void setServerCN( final String serverCN) { this.serverCN = serverCN; }
+
+	//=================================================================================================
+	// assistant methods
+
+	//-------------------------------------------------------------------------------------------------
+	@Override
+	public void doFilter(final ServletRequest request, final ServletResponse response, final FilterChain chain) throws IOException, ServletException {
+		if (request instanceof HttpServletRequest) {
+			log.debug("Checking access in SubscriberNotificationAccessControlFilter...");
+			try {
+				final HttpServletRequest httpRequest = (HttpServletRequest) request;
+				final String requestTarget = Utilities.stripEndSlash(httpRequest.getRequestURL().toString());
+					
+				if ( eventTypeMap != null) {
+					
+					for (final String notifacationUri  : eventTypeMap.values()) {
```

Comment:
typo in loop varible name: notificationUri

Reply:
fixed